pkg/agent/controller/agent.go

```diff
@@ -89,10 +85,18 @@ func NewWithDetail(spec *AgentSpecification, syncerConf *broker.SyncerConfig, re
 		BrokerResourcesEquivalent: agentController.serviceImportEquivalent,
 	}
 
+	endpointSliceSyncer := broker.ResourceConfig{
+		LocalSourceNamespace:      metav1.NamespaceAll,
+		LocalResourceType:         &discovery.EndpointSlice{},
+		BrokerResourceType:        &discovery.EndpointSlice{},
+		BrokerResourcesEquivalent: agentController.endpointSliceEquivalent,
```

Comment:
Since you're syncing `EndpointSlice` <-> `EndpointSlice`, you want to specify the `LocalClusterID` which is on `broker.SyncerConfig` so you'll need a separate broker `Syncer`.

Reply:
In addition, if we had a `BrokerTransformer`  that set the desired target NS then we wouldn't need the `EndpointSliceController` to duplicate it. But for that we need a change in admiral to use the transformed NS if the `LocalNamespace` is "". The syncer config here would then become:

```
endpointSyncerConfig := &broker.SyncerConfig{
    LocalRestConfig:  syncerConf.LocalRestConfig,
    LocalNamespace:   metav1.NamespaceAll,
    LocalClusterID:  spec.ClusterID,
    BrokerRestConfig: syncerConf.BrokerRestConfig,
    BrokerNamespace:  syncerConf.BrokerNamespace,
    Scheme:          syncerConf.Scheme,
    ResourceConfigs:  []broker.ResourceConfig{
        {
            LocalSourceNamespace:      metav1.NamespaceAll,
	    LocalResourceType:         &discovery.EndpointSlice{},
	    BrokerResourceType:        &discovery.EndpointSlice{},
	    BrokerResourcesEquivalent: agentController.endpointSliceEquivalent,
	    BrokerTransform: remoteEndpointSliceToLocal,
	},
    },
}

```

Reply:
Done.